selfhost/typechecker.jakt

```diff
@@ -4435,6 +4438,155 @@ struct Typechecker {
         return CheckedStatement::If(condition: checked_condition, then_block: checked_block, else_statement: checked_else, span)
     }
 
+    // FIXME: Desugar of variable binding when using IS operator should probably be done in codegen stage
+    function expand_block_bindings(this, anon block: ParsedBlock, checked_condition: CheckedExpression, span: Span) throws -> ParsedBlock {
```

Comment:
I'm not quite sure what _precisely_ this desugars to, but you can directly codegen this, no need to involve a match
```
if foo is Bar(x: y) and expr {
    true_branch
}
```
(provided that `foo` is of an enum type)
```c++
if (auto _value = foo.get_pointer<FooDetails::Bar>()) {
    auto& y = _value.x;
    if (codegen.expr) {
        codegen.true_branch;
    }
}
```

for simplicity, `or` could simply be split up into multi `if` statements before entering the codegen stage.